src/libserialize/leb128.rs

```diff
@@ -9,33 +9,40 @@
 // except according to those terms.
 
 #[inline]
-pub fn write_to_vec(vec: &mut Vec<u8>, position: &mut usize, byte: u8) {
-    if *position == vec.len() {
+fn write_to_vec(vec: &mut Vec<u8>, position: usize, byte: u8) {
+    if position == vec.len() {
         vec.push(byte);
     } else {
-        vec[*position] = byte;
+        vec[position] = byte;
     }
-
-    *position += 1;
 }
 
-pub fn write_unsigned_leb128(out: &mut Vec<u8>, start_position: usize, mut value: u64) -> usize {
-    let mut position = start_position;
+#[inline]
+pub fn write_unsigned_leb128_to<W>(mut value: u64, mut write: W) -> usize
```

Comment:
Can you add documentation on what the requirements for `write` are? Something like "`write` is a function that writes a single byte to the output destination at the given position. It is used to write the leb128 value to positions [0..N), where N is the number of bytes the value takes up".